test/test_client.py

```diff
@@ -347,6 +347,16 @@ def test_metadata(self):
         options = client._MongoClient__options
         self.assertEqual(options.pool_options.metadata, metadata)
 
+    def test_container_metadata(self):
+        os.environ[ENV_VAR_K8S] = "1"
```

Comment:
Let's use mock.patch.dict to edit the environ that way it gets correctly reset to the previous value.